### PR TITLE
[RFC] Proposal to define ECS-compliance - Stage 1

### PR DESCRIPTION
**Pull request by ebeahan**
This RFC PR proposes the project establishes a definition of an "ECS-compliant" event.

Addresses #1780 

### Stage 1 Criteria

- [x] Opened pull request for this proposal
- [x] Identified "sponsor" at Elastic who will participate in RFC process and take ownership of the change after the process completes
- [x] ~Outlined initial field definitions~ - N/A
- [x] High-level description of examples of usage
- [x] High-level description of example sources of data
- [x] Identified potential concerns and implementation challenges/complexity
- [ ] Subject matter experts identified and weighed in on the high level utility of these changes in the pull request
- [x] ECS team weighed in on appropriateness of these changes in the pull request

### CODE REVIEW

File: rfcs/text/0000-definition-of-ecs-compliance.md [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+* use nested fields over dotted. ECS events should use nested objects, like `{ "log": { "level": "debug" }}`), over dotted field names, such as `{ "log.level": "debug" }`.
+
+### Recommended guidelines
+
+ECS-compliant events SHOULD:
+
+* map the contents of the original event to as many ECS fields as possible.
+* populate the top-level `message` field.
+* store the entire raw, original event in `event.original`. Indexing and doc_values are disabled on `event.original` to reduce store, but the value is retrieved from `_source`.
+* use an array if the field expects and array event if populating a single value.
```
Comment (kgeller) [inline rfcs/text/0000-definition-of-ecs-compliance.md:36]: This line confuses me a little bit. I think you're just trying to say use an array if we specify array, but I wanted to double check

Comment (ebeahan) [reply rfcs/text/0000-definition-of-ecs-compliance.md]: Rephrased in https://github.com/elastic/ecs/pull/1868/commits/fa68b3b21ced0b394a11f255b835ec6958e3e791. Is the rephrasing an improvement?

Comment (kgeller) [reply rfcs/text/0000-definition-of-ecs-compliance.md]: Yes! Thanks for the update

**Review by kgeller — APPROVED** [review]
LGTM

**Pull request merged**